### DISCUSSION TITLE
Screenshot Feature introduction & Guide

### DISCUSSION THREAD

**Discussion by jhslee** [naverz/zepeto-studio-global#15] [Official Samples]
Hello, we have prepared an example project and guide to feature the new Screenshot feature available as of world package version 1.1.5 😄

# You can Download the sample project through Github
[ScreenShotSample Github](https://github.com/naverz/zepeto-multiplay-example.git)

# Guide
![ScreenShotSample](https://user-images.githubusercontent.com/100118650/155677131-fee61cec-45af-4779-8db9-60513edf2b75.png)

This is a New World screenshot example project using image related functions of ZepetoWorldContent API. In order to run the project, you must [login to ZEPETO](https://studio.zepeto.me/us/guides/preparing-for-world-creating) in advance.

<br/><br/>

## Overview
This is an example written based on It consists of a selfie mode and a ZepetoCamera mode, and saves or shares the image as a RenderTexture, and uses the image to create a feed.
This is an example using the guide from [Image Saving/Sharing/Feed Posting](https://studio.zepeto.me/guides/image-save-share-upload). It utilizes the selfie mode and ZepetoCamera to save/share an image as a Render texture, and uses the image to create a feed.

<br/><br/>

## Guide

When applying to an existing project, its recommended to follow the guidelines below:

- **The version must be Zepeto.World 1.1.5 or greater**
    
If your world version is not 1.1.5 or higher you cannot use the ZepetoWorldContent function. You can proceed with version update in Window - PackageManager - Zepeto.World.
    
- **Layer 21 is designated as the player layer in the sample project**
    
If your project layer 21 is currently in use, please adjust the playerLayer value in the ZepetoScreenShotModule.ts

   ![ScreenShot_Setting1](https://user-images.githubusercontent.com/100118650/155678297-a3d18367-e31c-4779-91ef-ae62d3851989.png)

<br/>
- You must have ZepetoPlayers in your Scene. 

- **Player Intantiation code must be written separately**    
Please refer to PlayerCreator.ts in the sample project for more information<br/>

- **You must have ZepetoScreenShotModule in your scene.**<br/>
![ScreenShot_Setting2](https://user-images.githubusercontent.com/100118650/155678405-8065b5f4-2252-4601-800c-6d0bcd56e332.png)

- **You must use an animator with IKPass enabled. **<br/>
![ScreenShot_Setting3](https://user-images.githubusercontent.com/100118650/155678475-fd0fb7b9-9a49-4710-94c3-7022783c4324.png)

   ▶️ If utilizing an existing ZepetoAnimator: 
Copy the ZepetoAnimator object in ZepetoPlayers into Assets/... path in your local environment. Enable IKPass in the copied ZepetoAnimator, and replace the AnimatorController of the ZepetoPlayer component with the corresponding animator.

   ▶️ If Using a custom animator :
    Activate IKPass of the Animator you are currently using. .<br/><br/>

    
- **Screenshot feature supports only Horizontal**<br/>
Please set your orientation to Orientation - Horizontal in your ZepetoWorldSettings.<br/>
![ScreenShot_Setting4](https://user-images.githubusercontent.com/100118650/155678525-ec4f87dd-f294-4b4c-a8b6-20063c58eebc.png)

- **Camera Settings**<br/>
    
The ZepetoCamera mode settings are adjusted from ZepetoPlayers - Camera.
    
   ![ScreenShot_Setting5](https://user-images.githubusercontent.com/100118650/155679229-ed7687d3-9b84-4359-b1f2-72103dd3cdeb.png)

The selfie camera mode settings are adjusted from the SelfieCamera.ts script variables of the Resources - SelfieCamera prefab.

   ![ScreenShot_Setting6](https://user-images.githubusercontent.com/100118650/155679267-74711adb-363e-4454-8289-5cd7e7f778d5.png)

**Comment by vanshunzpt** [comment]
i'm confusion about it becasue there no video for explain it. btw what is client starter and client starter v2 ?
